### PR TITLE
make deferCount TINYINT + make MessageId more typed

### PR DESCRIPTION
**Pull request by VDBBjorn**

### CODE REVIEW

File: Rebus.Idempotency.MySql/MySqlMessageStorage.cs [hunk 1]
```diff
@@ -42,16 +42,16 @@ public async Task<MessageData> Find(MessageId messageId)
                             WHERE s.`message_id` = @message_id and s.`defer_count`= @defer_count
                         ";
                     command.Parameters.Add(command.CreateParameter("message_id", DbType.String, messageId.OriginalMessageId));
-                    command.Parameters.Add(command.CreateParameter("defer_count", DbType.Int32, messageId.DeferCount));
+                    command.Parameters.Add(command.CreateParameter("defer_count", DbType.SByte, messageId.DeferCount));
```
Comment (mario-d-s) [inline Rebus.Idempotency.MySql/MySqlMessageStorage.cs]: Doesn't this mean `signed`? This should be `unsigned` and I believe MySQL supports that...

File: Rebus.Idempotency.MySql/MySqlMessageStorage.cs [hunk 2]
```diff
@@ -215,7 +215,7 @@ public async Task EnsureTablesAreCreated()
                         $@"
                             CREATE TABLE `{_dataTableName}` (
                                 `message_id` CHAR(36) NOT NULL,
-                                `defer_count` INT NOT NULL DEFAULT 0,
+                                `defer_count` TINYINT(3) NOT NULL DEFAULT 0,
```
Comment (mario-d-s) [inline Rebus.Idempotency.MySql/MySqlMessageStorage.cs]:
```suggestion
                                `defer_count` TINYINT(3) UNSIGNED NOT NULL DEFAULT 0,
```

File: Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
-    <VersionPrefix>1.3.0</VersionPrefix>
+    <VersionPrefix>1.3.1</VersionPrefix>
```
Comment (mario-d-s) [inline Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj]: Given that the column specification is again different this feels like at least a minor number (if not major)

**Review by mario-d-s — CHANGES_REQUESTED** [review]
Please see remarks (making it an unsigned tinyint).
Additionally: has the ID column been widened for any table? If so, the GUID parsing might again give trouble I'm afraid...

File: Rebus.Idempotency.MySql/MySqlMessageStorage.cs [hunk 4]
```diff
@@ -42,16 +42,16 @@ public async Task<MessageData> Find(MessageId messageId)
...
                     try
                     {
                         using (var reader = await command.ExecuteReaderAsync())
                         {
                             if (!await reader.ReadAsync()) return null;
 
-                            var msgId = ((Guid)reader.ExtractValue("message_id")).ToString();
-                            var deferCount = (int?)reader.ExtractValue("defer_count");
+                            var msgId = (Guid)reader.ExtractValue("message_id");
+                            var deferCount = (int?)(sbyte)reader.ExtractValue("defer_count");
```
Comment (dietermijle) [inline Rebus.Idempotency.MySql/MySqlMessageStorage.cs]: Why do we keep the defer count as an int? Can't we avoid the double cast?

File: Rebus.Idempotency.MySql/MySqlMessageStorage.cs (hunk 2 above)
Comment (dietermijle) [reply Rebus.Idempotency.MySql/MySqlMessageStorage.cs]: Should be unsigned indeed! :+1:

File: Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj (hunk 3 above)
Comment (dietermijle) [reply Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj]: As we haven't really released this, its ok to keep it as 1.3?

**Review by dietermijle — CHANGES_REQUESTED** [review]

File: Rebus.Idempotency/MessageId.cs [hunk 5]
```diff
@@ -7,9 +7,9 @@ public class MessageId
         public Guid OriginalMessageId;
         public int DeferCount = 0;
```
Comment (dietermijle) [inline Rebus.Idempotency/MessageId.cs]: Lets reflect the UNSIGNED byte field here in code and change this to `byte`.

**Review by dietermijle — CHANGES_REQUESTED** [review]

File: Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj (hunk 3 above)
Comment (mario-d-s) [reply Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj]: @dietermijle I have pushed these packages before a couple of weeks back when Bjorn was on leave that friday, to try and move things forward. But yeah we are the only ones looking into it now so it's fine I guess...

Comment (VDBBjorn) [reply Rebus.Idempotency.MySql/Rebus.Idempotency.MySql.csproj]: I kept is as 1.3.0 and will overwrite the package that was already published (for once)

**Review by dietermijle — APPROVED** [review]

**Review by robvanpamel — APPROVED** [review]

**Pull request merged**